src/util/dev/server.ts

```diff
@@ -329,11 +347,70 @@ export default class DevServer {
       }
     }
 
+    const apiFiles = await getApiFiles(this.cwd, this.output);
+
+    if (apiFiles.length > 0) {
+      if (!config.builds || config.builds.length === 0) {
+        config.builds = [];
+
+        if (pkg) {
+          const staticBuilder = await detectBuilder(pkg);
+
+          if (staticBuilder) {
+            config.builds.push(staticBuilder);
+          }
+        }
+
+        if (apiFiles && apiFiles.length > 0) {
+          const apiBuilds = await detectApiBuilders(apiFiles);
+
+          if (apiBuilds && apiBuilds.length > 0) {
+            config.builds.push(...apiBuilds);
+          }
+        }
+      }
+
+      const { defaultRoutes, error } = await detectApiRoutes(apiFiles);
```

Comment:
I just want to point out that returning `error` here seems a bit inconsistent with our other cases of error handling. Is this a convention we want to move towards using more of in the future?

Reply:
The API I implemented this first used this convention so I adapted it. And I've seen some  packages use it as well